conda_libmamba_solver/solver.py

```diff
@@ -105,12 +105,14 @@ def __init__(
         for spec in specs_to_add:
             if isinstance(spec, PackageRecord):
                 spec = MatchSpec(str(spec))
-            else:
+            elif isinstance(spec, MatchSpec):
                 spec_str = str(spec)
                 if "::" in spec_str:
                     for arg in sys.argv:
                         if spec_str in arg:
-                            spec = MatchSpec(arg)
+                            ms_from_arg = MatchSpec(arg)
```

Comment:
I checked this PR when running

```shell
conda create -n test -d conda-forge::jax conda-forge::jaxlib
```

specs_to_add is of type `List[str]` so the `elif isinstance(spec, MatchSpec):` branch is not touched. Is this intended? If so I am wondering why you are modifying lines 114 & 115.

Reply:
Ah, good catch! I think it might be the case in some tests that are passing `List[MatchSpec]` directly. We'll have a clearer view when #223 is merged.